src/sonic_ax_impl/mibs/ietf/rfc2737.py

```diff
@@ -0,0 +1,103 @@
+from enum import Enum, unique
+from bisect import bisect_right
+
+from swsssdk import ConfigDBConnector
+from ax_interface import MIBMeta, MIBUpdater, ValueType, SubtreeMIBEntry
+
+
+@unique
+class PhysicalClass(int, Enum):
+    """
+    Physical classes defined in RFC 2737.
+    """
+    OTHER = 1
+    UNKNOWN = 2
+    CHASSIS = 3
+    BACKPLANE = 4
+    CONTAINER = 5
+    POWERSUPPLY = 6
+    FAN = 7
+    SENSOR = 8
+    MODULE = 9
+    PORT = 10
+    STACK = 11
+
+
+class PhysicalTableMIBUpdater(MIBUpdater):
+
+    DEVICE_METADATA = "DEVICE_METADATA"
+    CHASSIS_ID = 1
+
+    def __init__(self):
+        super().__init__()
+
+        self.configdb = ConfigDBConnector()
+        self.configdb.connect()
+
+        self.physical_classes = []
+        self.physical_classes_map = {}
+
+        self.reinit_data()
+
+    def reinit_data(self):
+        """
+        Re-initialize all data.
+        """
+        device_metadata = self.configdb.get_table(self.DEVICE_METADATA)
```

Comment:
>device_metadata [](start = 8, length = 15)

What happens if not exist? What happens if ill-formatted?

Reply:
Done. Added checks